apps/arweave/src/ar_data_sync.erl

```diff
@@ -199,11 +206,44 @@ add_chunk(DataRoot, DataPath, Chunk, Offset, TXSize) ->
 							ets:update_counter(ar_data_sync_state, disk_pool_size,
 									{2, ChunkSize}),
 							prometheus_gauge:inc(pending_chunks_size, ChunkSize),
-							ok
+							case is_estimated_long_term_chunk(DataRootOffsetReply, EndOffset3) of
+								false ->
+									temporary;
+								true ->
+									ok
+							end
 					end
 			end
 	end.
 
+%% @doc Return true if we expect the chunk with the given data root index value and
+%% relative end offset to end up in one of the configured storage modules.
+is_estimated_long_term_chunk(DataRootOffsetReply, EndOffset) ->
+	IsTipCovered = is_tip_covered(),
+	case DataRootOffsetReply of
+		not_found ->
+			%% A chunk from a pending transaction.
+			IsTipCovered;
+		{ok, {TXStartOffset, _}} ->
+			WeaveSize = ar_node:get_current_weave_size(),
+			Size = ar_node:get_recent_max_block_size(),
+			AbsoluteEndOffset = TXStartOffset + EndOffset,
+			case AbsoluteEndOffset > WeaveSize - Size * 4 of
```

Comment:
Why do we use `Size * 4` here and `Size * 2` in `is_tip_covered`? Like if AbsoluteEndOffset is at `WeaveSize - Size * 3` and `is_tip_covered` is `true` - but we only cover `Weave - Size * 2`... would that cause a problem?

Reply:
Hm, good point, the offsets between `-Size*4` and `-Size*2` would be excluded for no good reason. [This](https://github.com/ArweaveTeam/arweave/pull/537/commits/deb70d9e6e446b97817f32b9483853e6487e0c4c) makes more sense. `4` is the constant that expresses the confidence a reorg won't change the offset whereas `2` expresses the expected conservative offset "drift" in the mempool.

Reply:
looks good!